presto-hive/src/main/java/com/facebook/presto/hive/HiveWriteUtils.java

```diff
@@ -376,8 +377,13 @@ public static void renameDirectory(HdfsEnvironment hdfsEnvironment, String schem
     public static Path createTemporaryPath(HdfsEnvironment hdfsEnvironment, Path targetPath)
     {
         // use a per-user temporary directory to avoid permission problems
-        // TODO: this should use Hadoop UserGroupInformation
-        String temporaryPrefix = "/tmp/presto-" + StandardSystemProperty.USER_NAME.value();
+        String temporaryPrefix;
+        try {
+            temporaryPrefix = "/tmp/presto-" + UserGroupInformation.getCurrentUser().getUserName();
```

Comment:
@billonahill you can give this a try: https://github.com/Teradata/presto/pull/109

Also if you could explain in some more datail the problem you were facing - so I could put that in commit message - it would be helpful.
The old code resolved `/tmp` in context of `targetPath`. So there should not be file system mismatch. The problem which could appear was that there was no `/tmp` on the target file system. Is that what you saw?